shared/protos/treenode.proto

```diff
@@ -16,24 +16,26 @@ import "Settings.proto";
 
 message TreeNode {
   oneof type {
-    bool folder = 1;
+    bool folder = 100;
+    // NOTE: these are all off by 1 cause protobutt don't allow you to set field number to 0
+    resources.Object object = 1;
     resources.Sprite sprite = 2;
     resources.Sound sound = 3;
-    resources.Background background = 4;
-    resources.Path path = 5;
-    resources.Script script = 6;
-    resources.Shader shader = 7;
+    resources.Room room = 4;
+    resources.Background background = 5;
+    resources.Path path = 6;
+    resources.Script script = 7;
```

Comment:
Why did you rearrange these? They were in tree order before, and I'm quite certain it will effect RGM's default tree order.

Reply:
josh told me to use gm's indexes here and generate enum from this

Reply:
That's fine, but we can still use the order of the GM tree so RGM's default tree is in the same order. That generation goes by field declaration order, not the ID order.